app/controllers/concerns/hyrax/breadcrumbs.rb

```diff
@@ -3,12 +3,18 @@ module Breadcrumbs
     extend ActiveSupport::Concern
 
     def build_breadcrumbs
-      if request.referer
-        trail_from_referer
-      else
-        default_trail
-        add_breadcrumb_for_controller if user_signed_in?
-        add_breadcrumb_for_action
+      return if build_breadcrumbs_skip
+      trail_from_referer
+    end
+
+    def build_breadcrumbs_skip
+      return true if request.referer.blank?
+      referer_path = URI(request.referer).path
+      begin
+        Rails.application.routes.recognize_path(referer_path)
```

Comment:
To make sure I'm following the logic, if Rails recognizes the referer path, then we don't want to skip breadcrumbs.  Otherwise, we do want to skip them.

Reply:
Jeremy,
Right. The requirements are that breadcrumbs are displayed from internal references, but not from external references.
Thanks!
Fritz.